lib/pages-router.js

```diff
@@ -0,0 +1,162 @@
+            if (uriMatcher.match(uriPattern)) {
+                return {
+                    page: lookupTable.pages[uriPagesMap[uriPattern]],
+                    uriParams: uriMatcher.elements()
+                };
+            }
+        }
+        // No page found
+        return null;
+    }
+
+    /**
+     * Returns the URI of the login page specified in application configurations file.
+     * @param renderingContext {RenderingContext} application configurations
+     * @param lookupTable {LookupTable} lookup table
+     * @return {string} page URI.
+     */
+    function getLoginPageUri(renderingContext, lookupTable) {
+        var appData = renderingContext.app;
+        var loginPageFullName = appData.conf[constants.APP_CONF_LOGIN_PAGE];
+        if (loginPageFullName) {
+            var loginPage = lookupTable.pages[loginPageFullName];
+            if (loginPage) {
+                loginPage = utils.getFurthestChild(loginPage);
+                if (loginPage.disabled) {
+                    log.warn("Login page '" + loginPage.fullName + " mentioned in application "
+                             + "configuration file '" + constants.FILE_APP_CONF + "' is disabled.");
+                } else {
+                    return (appData.context + loginPage.definition[constants.PAGE_DEFINITION_URI]);
+                }
+            } else {
+                log.error("Login page '" + loginPageFullName + " mentioned in application "
+                          + "configuration file '" + constants.FILE_APP_CONF
+                          + "' does not exists.");
+            }
+        }
+        return (appData.context + "/");
+    }
+
+    /**
+     * Whether the specified page is processable or not.
+     * @param page {UIComponent} page
+     * @param renderingContext {RenderingContext} rendering context
+     * @param lookupTable {LookupTable} lookup table
+     * @param request {Object} HTTP request
+     * @param response {Object} HTTP response
+     * @return {boolean} <code>true</code> if processable, otherwise <code>false</code>
+     */
+    function isPageProcessable(page, renderingContext, lookupTable, request, response) {
+        if (page.disabled) {
+            // This page is disabled.
+            response.sendError(404, "Requested page '" + renderingContext.pageUri
+                                    + "' does not exists.");
+            return false;
+        }
+
+        var pageDefinition = page.definition;
+        if (utils.parseBoolean(pageDefinition[constants.UI_COMPONENT_DEFINITION_IS_ANONYMOUS])) {
+            // This is an anonymous page. So no need for an user session or checking permissions.
+            return true;
+        }
+
+        // This is not an anonymous page.
+        var user = renderingContext.user;
+        if (user) {
+            // An user has logged in.
+            var pagePermissions = pageDefinition[constants.UI_COMPONENT_DEFINITION_PERMISSIONS];
```

Comment:
this permission model should align with https://docs.wso2.com/display/IS500/Configuring+Roles+and+Permissions+for+a+Service+Provider